packages/hooks/src/useReactive/index.ts

```diff
@@ -25,7 +25,9 @@ function observer<T extends Record<string, any>>(initialVal: T, cb: () => void):
   const proxy = new Proxy<T>(initialVal, {
     get(target, key, receiver) {
       const res = Reflect.get(target, key, receiver);
-      return isObject(res) ? observer(res, cb) : res;
+      const isReactElement = isValidElement(res);
```

Comment:
It might not be the issue of React Element since its `_store` is just set to non-writable and non-configurable.

But this reproduction below still throws. So I suggest following the error to make a workaround for accessing objects which are non configurable and readonly above.

```tsx
const NonConfigurableObject = function createNonProxyableObject(): {user: {name: string}} {
  const obj = {}
  Reflect.defineProperty(obj, 'user', { 
    value: { name: "Alice" }, 
    writable: false, 
    configurable: false,
  })
  // Reflect.getOwnPropertyDescriptor(obj, 'user')
  // { value: {…}, writable: false, enumerable: true, configurable: false }
  return obj as any
}();

export default () => {
  const state = useReactive(NonConfigurableObject);

  return (
    <div>
      <div>{state.user.name}</div>
    </div>
  );
};
```

Reply:
Yeh, I realize this, so I am working on it  :)

Reply:
This situation may not be resolved temporarily, so this PR is used to solve other issue: https://github.com/alibaba/hooks/issues/2080

Reply:
For the problem of "non-writable and non-configurable", I've tried a lot of methods, but no matter which method, there are always some details that cannot be processed.

The best method I've tried is "clone the initialState deeply", example:

```ts
import { cloneDeepWith } from 'lodash';

function deepClone<S>(initialState: S): S {
  const copy = cloneDeepWith(initialState, (value) => {
    const ret = {};

    for (const key of Reflect.ownKeys(value)) {
      ret[key] = Reflect.get(value, key);
    }

    return ret;
  });

  return copy;
}
```

The code above can handles most situations, expect for the "getter/setter + this" in the prototype:

```ts
const parent: Record<string, string> = {
    name: 'parent',
    get value() {
      return this.name;
    },
  };

  const child: Record<string, string> = {
    name: 'child',
  };

  const { result } = renderHook(() => useReactive(parent));
  const proxy = result.current;

  Object.setPrototypeOf(child, proxy);

  expect(child.value).toBe('child');
  expect(proxy.value).toBe('parent');
  expect(parent.value).toBe('parent');

  act(() => delete child.name);
  expect(child.value).toBe('parent');
  expect(proxy.value).toBe('parent');
  expect(parent.value).toBe('parent');

  act(() => delete proxy.name);
  expect(child.value).toBeUndefined();
  expect(proxy.value).toBeUndefined();
  expect(parent.value).toBeUndefined();
```

The above test case fails.

Finally, I gave up trying and didn't find any perfect way.